applications/IgaApplication/custom_elements/iga_truss_element.cpp

```diff
@@ -0,0 +1,182 @@
+/*
+//  KRATOS  _____________
+//         /  _/ ____/   |
+//         / // / __/ /| |
+//       _/ // /_/ / ___ |
+//      /___/\____/_/  |_| Application
+//
+//  Main authors:   Anna Bauer
+//                  Thomas Oberbichler
+//                  Tobias Teschemacher
+*/
+
+// System includes
+#include "includes/define.h"
+#include "includes/variables.h"
+
+// External includes
+
+// Project includes
+#include "iga_truss_element.h"
+#include "iga_application_variables.h"
+
+namespace Kratos {
+
+Element::Pointer IgaTrussElement::Create(
+    IndexType NewId,
+    NodesArrayType const& ThisNodes,
+    PropertiesType::Pointer pProperties) const
+{
+    auto geometry = GetGeometry().Create(ThisNodes);
+
+    return Kratos::make_shared<IgaTrussElement>(NewId, geometry,
+        pProperties);
+}
+
+void IgaTrussElement::GetDofList(
+    DofsVectorType& rElementalDofList,
+    ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    rElementalDofList.resize(NumberOfDofs());
+
+    for (std::size_t i = 0; i < NumberOfNodes(); i++) {
+        SetDof(rElementalDofList, i, 0, DISPLACEMENT_X);
```

Comment:
no 2D version?

Reply:
At the moment I do not have time to develop 2D elements in parallel. Of course it would be possible to add a 2D version and extend the naming scheme to `Iga...Element2D`.

Another way to optimize the elements for specific situations would be a fixed number of nodes (e.g. `Iga...Element4N3D` for a cubic element, `Iga...ElementXN3D` for the default get-node-number-and-degree-at-runtime element). I made some tests without significant speedups.

About the class names: I think 'IgaElement' was suggested during our last discussion by @pooyan-dadvand. After some considerations I really like the solution.

Main reasons:
- Short and meaningful
- You can use other shapefunctions as long as you use the same shapefunctions for geometry and displacement (IsoGeometric) so it will always be some kind of IGA Element.
- If someone is interested in the IGA capabilities of Kratos he will probably look for 'Iga...' stuff
- Inside publications we can call them however we want